src/pulselistener/pulselistener/mercurial.py

```diff
@@ -107,6 +120,7 @@ def clean(self):
         - trigger push-to-try
         '''
         logger.info('Received diff {phid}'.format(**diff))
+        await asyncio.sleep(2)  # allow other tasks to run
```

Comment:
Do we need this because hglib blocks the event loop by waiting on the mercurial process?

Reply:
Exactly

Reply:
OK, that's a pity, it's quite hacky :P